[export] Preserve custom metadata for tensor constants

yiming0416 opened the pull request:
Summary:
Fixes https://github.com/pytorch/pytorch/issues/151476
The `custom_meta` collected from `mod` has keys that follow name of nodes in `mod`, which are inconsistent with the node names after the naming pass. For example a constant `b` will become `c_b`.

Test Plan: buck2 run caffe2/test:test_export -- -r test_run_decompositions_keep_tensor_constant_metadata

Differential Revision: D73703068

yiming0416 commented on torch/_export/utils.py:
```diff
@@ -1007,7 +1013,7 @@ def _extract_pytree_key(x):
             assert node.name in name_map
             node.name = node.target = name_map[node.name]
-            if node.name in custom_meta:
+            if node.name in custom_meta and node.meta.get("custom") is None:
```
@angelayi  The custom metadata of some nodes (e.g., parameters) have already been preserved. So here we only retrieve the assign to nodes that don't have custom metadata

angelayi replied:
Just to confirm, the metadata is the same though?

yiming0416 replied:
> Just to confirm, the metadata is the same though?

Added assertion to make sure other custom metadata is the same

yiming0416 committed:
[export] Preserve custom metadata for tensor constants (#152241)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/152241

Fixes https://github.com/pytorch/pytorch/issues/151476
The `custom_meta` collected from `mod` has keys that follow name of nodes in `mod`, which are inconsistent with the node names after the naming pass. For example a constant `b` will become `c_b`.

Test Plan: buck2 run caffe2/test:test_export -- -r test_run_decompositions_keep_tensor_constant_metadata

Differential Revision: D73703068
